middle_end/flambda2/from_lambda/lambda_to_flambda_primitives.ml

```diff
@@ -544,6 +544,62 @@ let bytes_like_set_safe ~dbg ~size_int ~access_size kind bytes index new_value =
         (bytes_like_set_unsafe ~access_size Bigstring bytes index new_value)
       bytes index
 
+(* Array vector load/store *)
+
+let array_vector_access_validity_condition array
+    (vec_kind : Lambda.boxed_vector) (array_kind : P.Array_kind.t) index =
+  P.Array_kind.check_vector_access_kind array_kind vec_kind;
+  let width =
+    match vec_kind with
+    | Pvec128 (Float64x2 | Int64x2) -> 1
+    | Pvec128 Int32x4 -> 3
+    | Pvec128 (Unknown128 | Float32x4 | Int16x8 | Int8x16) ->
+      (* CR mslater: replace once we have unboxed float32/int16/int8 *)
+      assert false
+  in
+  let width = H.Simple (Simple.const_int (Targetint_31_63.of_int width)) in
+  [ H.Binary
+      ( Int_comp (Tagged_immediate, Yielding_bool (Lt Unsigned)),
+        index,
+        Prim
+          (Binary
+             ( Int_arith (I.Tagged_immediate, Sub),
+               Prim (Unary (Array_length (Array_kind array_kind), array)),
```

Comment:
you're missing a `max_with_zero` call here to avoid underflow of the array length computation, similar to what is done in the `actual_max_length_for_string_like_access` function

Reply:
Fixed